src/Libraries/Microsoft.Extensions.Resilience/Resilience/Internal/ResilienceEnricher.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Diagnostics.ExceptionSummarization;
+using Microsoft.Extensions.Http.Telemetry;
+using Microsoft.Extensions.Options;
+using Polly.Extensions.Telemetry;
+
+namespace Microsoft.Extensions.Resilience.Resilience.Internal;
+
+internal class ResilienceEnricher
+{
+    private readonly FrozenDictionary<Type, Func<object, FailureResultContext>> _faultFactories;
+    private readonly IOutgoingRequestContext? _outgoingRequestContext;
+    private readonly IExceptionSummarizer _exceptionSummarizer;
+
+    public ResilienceEnricher(
+        IOptions<FailureEventMetricsOptions> metricsOptions,
+        IEnumerable<IOutgoingRequestContext> outgoingRequestContext,
+        IExceptionSummarizer exceptionSummarizer)
+    {
+        _faultFactories = metricsOptions.Value.Factories.ToFrozenDictionary();
+        _outgoingRequestContext = outgoingRequestContext.FirstOrDefault();
+        _exceptionSummarizer = exceptionSummarizer;
+    }
+
+    public void Enrich(EnrichmentContext context)
```

Comment:
Does `context` need a null check?

Reply:
Nope, it will never be null. Polly infra makes sure of that.

Reply:
Adding `_ = Throw.IfNull(context);` shouldn't hurt.

Reply:
We generally reserve Throw.IfNull for public methods. Internally within the context of an assembly, we have the C# compiler's null rules that protect us.

Reply:
Yeah, the validation is arbitrary as the null won't ever come here.